### PR TITLE
SRX-ISJDVO Add button functionality in release dialog

### PR DESCRIPTION
**Pull request by jaypontes**

### CODE REVIEW

**Comment by leandro-freiheit** [comment]
I think you should create tests for the button you added, plus for the functions you added (`createAppLock` and `deleteAppLock`).

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss [hunk 1]
```diff
@@ -86,18 +86,21 @@
...
-          }
-
-          .env-card-env-locks:nth-child(2) {
-            margin-left: 55px;
+          .env-card-label {
             display: flex;
-            flex-direction: row;
+
+            svg.env-card-env-lock-icon {
+              width: 24px;
```
Comment (leandro-freiheit) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.scss:93]: Don't leave values as these ones in the code. You can see these standards in the knowledge base ([here](https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/WORKSPACE_DOCUMENT/-Mn5HLuHv2wjoq0QKPax/)) Add these variables to the .scss variables in the `_variables.scss` ([here](https://github.com/freiheit-com/kuberpult/blob/main/services/frontend-service/src/assets/_variables.scss))

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 2]
```diff
@@ -127,26 +127,79 @@ export const calculateDistanceToUpstream = (envs: Environment[]): EnvSortOrder =
...
+        });
+    }, [app, env.name]);
+    return (
+        <li key={env.name} className={classNames('env-card', className)}>
+            <div className="env-card-header">
+                <Chip
+                    className={'release-environment'}
+                    label={
+                        <div className={classNames('env-card-label', className)}>
+                            env.name{' '}
```
Comment (mouhsen-ibrahim) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:202]: put env.name between {}

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 3]
```diff
@@ -127,26 +127,79 @@ export const calculateDistanceToUpstream = (envs: Environment[]): EnvSortOrder =
...
+}> = ({ env, envPrioMap, app, release, className }) => {
+    const deploy = useCallback(
+        () =>
+            addAction({
+                action: {
+                    $case: 'deploy',
+                    deploy: {
+                        environment: env.name,
+                        application: app,
+                        version: release.version,
```
Comment (mouhsen-ibrahim) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:172]: Here release.version is undefined

**Review by mouhsen-ibrahim — CHANGES_REQUESTED** [review]
Please fix functionality of deploy button and add tests

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx (hunk 2 above)
Comment (jaypontes) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx]: Fixed!

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx (hunk 3 above)
Comment (jaypontes) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:171]: Seems to work for me. It's either an edge case I'm missing, or I suspect it's an issue with the `repository_checkedout` directory

**Comment by mouhsen-ibrahim** [comment]
@leandro-freiheit `createAppLock` and `deleteAppLock` are callbacks that call the addAction function which is already tested, no need to add tests just for them

**Review by mouhsen-ibrahim — APPROVED** [review]

**Pull request merged**